src/Bicep.Core/Features/FeatureProvider.cs

```diff
@@ -40,6 +40,8 @@ public FeatureProvider(RootConfiguration configuration)
 
         public static TraceVerbosity TracingVerbosity => ReadEnumEnvvar<TraceVerbosity>("BICEP_TRACING_VERBOSITY", TraceVerbosity.Basic);
 
+        public bool DynamicTypeLoading => configuration.ExperimentalFeaturesEnabled.DynamicTypeLoading ?? false;
```

Comment:
nit: We may want to rename it to `DynamicTypeLoadingEnabled`. The `Enabled` suffix is not needed for `ExperimentalFeatureEnabled.DynamicTypeLoading`, but I think it makes sense in the context of `FeatureProvider`.